### ISSUE TITLE
Compartment search should be able to search all compartment related resources with paramters valid for all resources.

### ISSUE THREAD

**Issue by d0roppe** [IBM/FHIR#1658]
**Is your feature request related to a problem? Please describe.**
Currently you have to specify a resource inside a compartment to be the basis for your search.  So there really isn't a patient compartment it is a Patient/Named Resource mini compartment.  

**Describe the solution you'd like**
This request is to be able to search on all resources inside a compartment.  Like be able to request all patient compartmtnet resource that were updated since a date. 
Like:  Get [base]/Patient/[id]/?_lastUpdated=ge2020-11-01

**Describe alternatives you've considered**

**Acceptance Criteria**
At least one acceptance criteria is included
In a defined compartment like patient compartment
be able to use any supported search parameter that is valid for all resource.
and have the search stay within the compartment only.
like Get [base]/Patient/[id]/?_lastUpdated=ge2020-11-01  will return all patient related resources that were last updated on November 1, 2020 or since then. 
1.  
  GIVEN [a precondition]
    AND [another precondition] WHEN [test step]
    AND [test step]
    THEN [verification step]
    AND [verification step]

**Additional context**

**Comment by lmsurpre** [comment]
I don't think this ever made it into the spec (or else it was in but got removed), but there is at least *some* prior art for this kind of thing:  https://fhirblog.com/2014/12/03/contemplating-compartments-in-fhir/
